src/js/models/layer.js

```diff
@@ -680,7 +683,9 @@ define(function (require, exports, module) {
                 vectorMaskEmpty: layerDescriptor.vectorMaskEmpty,
                 textWarningLevel: layerDescriptor.textWarningLevel,
                 isLinked: _extractIsLinked(layerDescriptor),
-                initialized: initialized || selected
+                boundsInitialized: layerDescriptor.boundsInitialized,
+                initialized: layerDescriptor.initialized,
+                descriptor: Immutable.Map(layerDescriptor)
```

Comment:
I suggest making this a private property, prefixed with an underscore. And I don't think there's any point in trying to convert it to be an immutable object? Also, just initializing an `Immutable.Map` with the object only shallowly converts the object, so this doesn't give you an immutable guarantee anyway.